src/device/tex.rs

```diff
@@ -300,55 +339,55 @@ pub enum FilterMethod {
     Anisotropic(u8)
 }
 
+/// The face of a cube texture to do an operation on.
+#[derive(Eq, Ord, PartialEq, PartialOrd, Hash, Copy, Clone, Debug)]
+#[allow(missing_docs)]
+pub enum CubeFace {
+    PosZ,
+    NegZ,
+    PosX,
+    NegX,
+    PosY,
+    NegY
+}
+
 /// Specifies how a given texture may be used. The available texture types are
 /// restricted by what Metal exposes, though this could conceivably be
 /// extended in the future. Note that a single texture can *only* ever be of
 /// one kind. A texture created as `Texture2D` will forever be `Texture2D`.
 // TODO: "Texture views" let you get around that limitation.
 #[derive(Eq, Ord, PartialEq, PartialOrd, Hash, Copy, Clone, Debug)]
-pub enum TextureKind {
+pub enum Kind {
     /// A single row of texels.
-    Texture1D,
+    D1,
```

Comment:
I was just summing you in gitter to ask what you think about it. `Dx` pattern may be a little confusing at first, but definitely very short to type.